platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/WorkNodeCodec.kt

```diff
@@ -64,13 +65,23 @@ class WorkNodeCodec(
         val nodeCount = nodes.size
         writeSmallInt(nodeCount)
         val scheduledNodeIds = HashMap<Node, Int>(nodeCount)
+        val entryNodes = ArrayList<Int>()
         nodes.forEach { node ->
             write(node)
-            scheduledNodeIds[node] = scheduledNodeIds.size
+            val nodeId = scheduledNodeIds.size
+            scheduledNodeIds[node] = nodeId
+            if (node.isEntryNode) {
```

Comment:
💭 Could we keep this information somewhere else, outside the nodes themselves? See next comment.